commands/command_pull.go

```diff
@@ -0,0 +1,34 @@
+package commands
+
+import (
+	"github.com/github/git-lfs/git"
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra"
+)
+
+var (
+	pullCmd = &cobra.Command{
+		Use:   "pull",
+		Short: "Downloads LFS files for the current ref, and checks out",
+		Run:   pullCommand,
+	}
+)
+
+func pullCommand(cmd *cobra.Command, args []string) {
+
+	ref, err := git.CurrentRef()
+	if err != nil {
+		Panic(err, "Could not pull")
+	}
+
+	// Previously we would only checkout files that were downloaded, as they
+	// were downloaded. However this would ignore files where the content was
+	// already present locally (since these are no longer included in transfer Q for
+	// better reporting purposes).
+	// So now we do exactly what we say on the tin, fetch then a separate checkout
```

Comment:
The problem is I don't know that it's even possible to use the old approach and be functionally complete as a checkout, it would ignore files that were already present in the store but not the working copy for example. Maybe you could parallelise the things that did need to be downloaded but you can't avoid having to check that there aren't things that didn't need to be downloaded (or were in multiple locations in the file system).